File: go/etl/etl.go
package etl

import "strings"

// Transform a scrabble score lookup system.
func Transform(scores map[int][]string) map[string]int {
    // Initialize a table in the new format with sufficient capacity for
    // the entire alphabet.
	result := make(map[string]int, 26)  

	for score, letters := range scores {
		for _, letter := range letters {
			result[strings.ToLower(letter)] = score
		}
	}

	return result
}

File: javascript/etl/etl.js
var transform = module.exports = function(scoresObj) {
  var result = {};
  Object.keys(scoresObj).forEach(function(scoreStr) {
    var score = +scoreStr;
    scoresObj[scoreStr].forEach(function(letter) {
      result[letter.toLowerCase()] = score;
    });
  });
  return result;
};
